File: examples/ruby/rails/db/migrate/20240101000000_create_tables.rb
class CreateTables < ActiveRecord::Migration[7.1]
  def change
    create_table :organizations, id: false do |t|
      t.string :id, primary_key: true, limit: 36, default: -> { "gen_random_uuid()::text" }
      t.string :name, null: false
      t.timestamps
    end

    create_table :users, id: false do |t|
      t.string :id, primary_key: true, limit: 36, default: -> { "gen_random_uuid()::text" }
      t.string :email, null: false
      t.string :name, null: false
      t.string :organization_id, limit: 36, null: false
      t.timestamps
    end
    add_index :users, :email, unique: true
    add_foreign_key :users, :organizations

    create_table :projects, id: false do |t|
      t.string :id, primary_key: true, limit: 36, default: -> { "gen_random_uuid()::text" }
      t.string :name, null: false
      t.string :organization_id, limit: 36, null: false
      t.timestamps
    end
    add_foreign_key :projects, :organizations

    create_table :tasks, id: false do |t|
      t.string :id, primary_key: true, limit: 36, default: -> { "gen_random_uuid()::text" }
      t.string :title, null: false
      t.string :status, null: false, default: "todo"
      t.string :organization_id, limit: 36, null: false
      t.string :project_id, limit: 36, null: false
      t.string :assignee_id, limit: 36, null: false
      t.timestamps
    end
    add_foreign_key :tasks, :organizations
    add_foreign_key :tasks, :projects
    add_foreign_key :tasks, :users, column: :assignee_id
  end
end

File: examples/php/laravel/database/migrations/2024_01_01_000000_create_tables.php
<?php

use Illuminate\Database\Migrations\Migration;
use Illuminate\Database\Schema\Blueprint;
use Illuminate\Support\Facades\Schema;

return new class extends Migration {
    public function up(): void
    {
        Schema::create('organizations', function (Blueprint $table) {
            $table->uuid('id')->primary()->default(DB::raw('gen_random_uuid()::text'));
            $table->string('name');
            $table->timestamps();
        });

        Schema::create('users', function (Blueprint $table) {
            $table->uuid('id')->primary()->default(DB::raw('gen_random_uuid()::text'));
            $table->string('email')->unique();
            $table->string('name');
            $table->string('organization_id', 36);
            $table->foreign('organization_id')->references('id')->on('organizations');
            $table->timestamps();
        });

        Schema::create('projects', function (Blueprint $table) {
            $table->uuid('id')->primary()->default(DB::raw('gen_random_uuid()::text'));
            $table->string('name');
            $table->string('organization_id', 36);
            $table->foreign('organization_id')->references('id')->on('organizations');
            $table->timestamps();
        });

        Schema::create('tasks', function (Blueprint $table) {
            $table->uuid('id')->primary()->default(DB::raw('gen_random_uuid()::text'));
            $table->string('title');
            $table->string('status')->default('todo');
            $table->string('organization_id', 36);
            $table->string('project_id', 36);
            $table->string('assignee_id', 36);
            $table->foreign('organization_id')->references('id')->on('organizations');
            $table->foreign('project_id')->references('id')->on('projects');
            $table->foreign('assignee_id')->references('id')->on('users');
            $table->timestamps();
        });
    }

    public function down(): void
    {
        Schema::dropIfExists('tasks');
        Schema::dropIfExists('projects');
        Schema::dropIfExists('users');
        Schema::dropIfExists('organizations');
    }
};
